torch/distributed/_composable/fully_shard.py

```diff
@@ -32,7 +33,7 @@ def fully_shard(
     process_group: Optional[dist.ProcessGroup] = None,
     mixed_precision: Optional[MixedPrecision] = None,
     cpu_offload: Optional[CPUOffload] = None,
-    auto_wrap_policy: Optional[Callable] = None,
+    policy: Optional[_FSDPPolicy] = None,
```

Comment:
This is composable FSDP. In my understanding, we should be able to change the constructor?

I wanted `policy` to be general because we can configure FSDP this way. This can be an entry point for different flavors of FSDP. One option to enable tensor shape preservation may be via `policy`.

Reply:
The wrapper `FullyShardedDataParallel` still calls it `auto_wrap_policy` in its constructor.

Reply:
oh I see, that sounds good then.